pkg/sbom/scanner/scanner.go

```diff
@@ -124,6 +124,10 @@ func (s *Scanner) enoughDiskSpace(opts sbom.ScanOptions) error {
 // sendResult sends a ScanResult to the channel associated with the scan request.
 // This function should not be blocking
 func sendResult(requestID string, result *sbom.ScanResult, collector collectors.Collector) {
+	if result == nil {
+		log.Errorf("nil result for '%s'", requestID)
+		return
+	}
```

Comment:
Can we instead avoid invoking `sendResult` with a `nil` `result` ?

I’m wondering if this condition can even happen with the current version of the code.
`sendResult` seems to be called only line 247 where `result` is
* either the non-`nil` return value of `checkDiskSpace` line 238 and line 241
* or the return value of `performScan` line 244.
And, given line 278, it seems that `perfomrScan` cannot return a `nil` pointer.

So, I feel that this block is dead code that can never be executed.

```suggestion
```

Reply:
Indeed it can never be executed but I prefer safe handling of pointers to avoid misusage in the future.